contracts/Everlong.sol

```diff
@@ -215,50 +239,95 @@ contract Everlong is IEverlong {
+        // and keep remaining target.
+        _closePositions(
+            _assets -
+                ERC20(_asset).balanceOf(address(this)) +
+                targetIdleLiquidity()
+        );
     }
 
     // ╭─────────────────────────────────────────────────────────╮
     // │ Rebalancing                                             │
     // ╰─────────────────────────────────────────────────────────╯
 
     /// @notice Rebalance the everlong portfolio by closing mature positions
-    ///         and using the proceeds to open new positions.
+    ///         and using the proceeds over target idle to open new positions.
     function rebalance() public override {
+        // Early return if no rebalancing is needed.
+        if (!canRebalance()) {
+            return;
+        }
+
         // Close matured positions.
         _closeMaturedPositions();
 
-        // Spend idle on opening a new position. Leave an extra wei for the
-        // approval to keep the slot warm.
-        uint256 toSpend = ERC20(_asset).balanceOf(address(this));
+        // Amount to spend is the current balance less the target idle.
+        uint256 toSpend = ERC20(_asset).balanceOf(address(this)) -
+            targetIdleLiquidity();
+
+        // Open a new position. Leave an extra wei for the approval to keep
+        // the slot warm.
         ERC20(_asset).forceApprove(address(hyperdrive), toSpend + 1);
         (uint256 maturityTime, uint256 bondAmount) = IHyperdrive(hyperdrive)
             .openLong(asBase, toSpend, "");
 
         // Account for the new position in the portfolio.
         _portfolio.handleOpenPosition(maturityTime, bondAmount);
+
+        emit Rebalanced();
     }
 
-    // FIXME: Consider idle liquidity + maybe maxLong?
-    //
-    /// @notice Returns whether the portfolio needs rebalancing.
-    /// @return True if the portfolio needs rebalancing, false otherwise.
+    /// @notice Returns true if the portfolio can be rebalanced.
+    /// @notice The portfolio can be rebalanced if:
+    ///         - Any positions are matured.
+    ///         - The current idle liquidity is above the target.
+    /// @return True if the portfolio can be rebalanced, false otherwise.
     function canRebalance() public view returns (bool) {
-        return true;
+        return
+            (!_portfolio.isEmpty() &&
+                IHyperdrive(hyperdrive).isMature(_portfolio.head())) ||
+            ERC20(_asset).balanceOf(address(this)) > targetIdleLiquidity();
+    }
+
+    // TODO: Use cached poolconfig
+    //
+    /// @notice Returns the target amount of funds to keep idle in Everlong.
+    /// @dev If the target amount is lower than Hyperdrive's minimum,
+    ///      then Hyperdrive's minimum becomes the target.
+    /// @return assets Target amount of idle assets.
+    function targetIdleLiquidity() public view returns (uint256 assets) {
+        assets = targetIdleLiquidityPercentage
+            .mulDivDown(totalAssets(), ONE)
```

Comment:
`mulDivDown` is given by the following calculation:

```
a.mulDivDown(b, c) = (a * b) / c
```

`mulDown` is given by the following calculation:

```
a.mulDown(b) = (a * b) / ONE
```

where `ONE = 1e18` for our `FixedPointMath` library. 

With this in mind, you can replace this with `mulDown`.